tests/components/sunweg/test_config_flow.py

```diff
@@ -40,12 +40,97 @@ async def test_incorrect_login(hass: HomeAssistant) -> None:
     assert result["errors"] == {"base": "invalid_auth"}
 
 
+async def test_server_unavailable(hass: HomeAssistant) -> None:
+    """Test when the SunWEG server don't respond."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch.object(
+        APIHelper, "authenticate", side_effect=SunWegApiError("Internal Server Error")
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], SUNWEG_USER_INPUT
+        )
+
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {"base": "timeout_connect"}
+
+
+async def test_reauth(hass: HomeAssistant) -> None:
```

Comment:
Can we also add a test which shows that a reauth flow is triggered when `api.authenticate` fails, for example like this?
https://github.com/home-assistant/core/blob/c78e089526076a6ede9bed02fb67d17182e26d4f/tests/components/roborock/test_init.py#L125-L138